src/lib.rs

```diff
@@ -36,6 +36,22 @@ app.add_plugins(MinimalPlugins).add_plugins(
         .disable::<ClientPlugin>()
         .set(ServerPlugin { tick_policy: TickPolicy::MaxTickRate(60) }),
 );
+#
+# #[derive(Debug)]
+# struct X;
+# app.add_server_event_with::<X, _, _>(||{}, ||{});
+# app.update();
+#
+# let mut app = App::new();
+# app.add_plugins(MinimalPlugins).add_plugins(
+#    ReplicationPlugins
+#        .build()
+#        .disable::<ServerPlugin>()
+#        .set(ServerPlugin { tick_policy: TickPolicy::MaxTickRate(60) }),
+# );
+#
+# app.add_server_event_with::<X, _, _>(||{}, ||{});
+# app.update();
```

Comment:
Could you move into unit test to `server_event` module and add the same for `client_event`?